From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Manu Gupta <mgpta@amazon.com>
Date: Tue, 12 Oct 2021 14:55:52 -0700
Subject: [PATCH] --EKS-PATCH-- ConsistentRead tries 10 times

Taken from commit on upstream Kubernetes:
https://github.com/kubernetes/kubernetes/pull/102059/commits/82cfe9f14f8fb445d682ce2774ea44ce54885e81.
This commit is part of upstream PR
https://github.com/kubernetes/kubernetes/pull/102059, which is part
of 1.22.

From the original commit message:
    We've seen clusters where 3 attempts were not enough. Bumping to
    10. The slowdown should be negligible and it will reduce retry
    attempts in the upper layers of kubelet.

Signed-off-by: Kirsten Schumy <ksschumy@amazon.com>
---
 staging/src/k8s.io/mount-utils/mount_helper_unix.go | 2 +-
 vendor/k8s.io/utils/mount/mount_helper_unix.go      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/staging/src/k8s.io/mount-utils/mount_helper_unix.go b/staging/src/k8s.io/mount-utils/mount_helper_unix.go
index 2c14a27c719..f3658647eea 100644
--- a/staging/src/k8s.io/mount-utils/mount_helper_unix.go
+++ b/staging/src/k8s.io/mount-utils/mount_helper_unix.go
@@ -32,7 +32,7 @@ const (
 	// At least number of fields per line in /proc/<pid>/mountinfo.
 	expectedAtLeastNumFieldsPerMountInfo = 10
 	// How many times to retry for a consistent read of /proc/mounts.
-	maxListTries = 3
+	maxListTries = 10
 )
 
 // IsCorruptedMnt return true if err is about corrupted mount point
diff --git a/vendor/k8s.io/utils/mount/mount_helper_unix.go b/vendor/k8s.io/utils/mount/mount_helper_unix.go
index 11b70ebc298..6110e7b2b35 100644
--- a/vendor/k8s.io/utils/mount/mount_helper_unix.go
+++ b/vendor/k8s.io/utils/mount/mount_helper_unix.go
@@ -32,7 +32,7 @@ const (
 	// At least number of fields per line in /proc/<pid>/mountinfo.
 	expectedAtLeastNumFieldsPerMountInfo = 10
 	// How many times to retry for a consistent read of /proc/mounts.
-	maxListTries = 3
+	maxListTries = 10
 )
 
 // IsCorruptedMnt return true if err is about corrupted mount point
